src/core/ws_pool.ml

```diff
@@ -14,46 +14,39 @@ module Id = struct
   let equal : t -> t -> bool = ( == )
 end
 
-type around_task = AT_pair : (t -> 'a) * (t -> 'a -> unit) -> around_task
-
-type task_full =
-  | T_start of {
-      ls: Task_local_storage.t;
-      f: task;
-    }
-  | T_resume : {
-      ls: Task_local_storage.t;
-      k: 'a -> unit;
-      x: 'a;
-    }
-      -> task_full
-
-type worker_state = {
-  pool_id_: Id.t;  (** Unique per pool *)
-  mutable thread: Thread.t;
-  q: task_full WSQ.t;  (** Work stealing queue *)
-  mutable cur_ls: Task_local_storage.t option;  (** Task storage *)
-  rng: Random.State.t;
-}
-(** State for a given worker. Only this worker is
-    allowed to push into the queue, but other workers
-    can come and steal from it if they're idle. *)
-
 type state = {
   id_: Id.t;
+      (** Unique per pool. Used to make sure tasks stay within the same pool. *)
   active: bool A.t;  (** Becomes [false] when the pool is shutdown. *)
-  workers: worker_state array;  (** Fixed set of workers. *)
-  main_q: task_full Queue.t;
+  mutable workers: worker_state array;  (** Fixed set of workers. *)
+  main_q: WL.task_full Queue.t;
       (** Main queue for tasks coming from the outside *)
   mutable n_waiting: int; (* protected by mutex *)
   mutable n_waiting_nonzero: bool;  (** [n_waiting > 0] *)
   mutex: Mutex.t;
   cond: Condition.t;
+  mutable as_runner: t;
+  (* init options *)
+  around_task: WL.around_task;
+  name: string option;
+  on_init_thread: dom_id:int -> t_id:int -> unit -> unit;
+  on_exit_thread: dom_id:int -> t_id:int -> unit -> unit;
   on_exn: exn -> Printexc.raw_backtrace -> unit;
-  around_task: around_task;
 }
 (** internal state *)
 
+and worker_state = {
+  mutable thread: Thread.t;
+  idx: int;
+  dom_id: int;
+  st: state;
+  q: WL.task_full WSQ.t;  (** Work stealing queue *)
```

Comment:
Comment unnecessary if we spell out WSQ :-)

Reply:
:p it'd be too long in the rest  of the module